nsxt/resource_nsxt_policy_lb_icmp_monitor_profile.go

```diff
@@ -0,0 +1,164 @@
+/* Copyright © 2020 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package nsxt
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/bindings"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/data"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/protocol/client"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/infra"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+)
+
+func resourceNsxtPolicyLBIcmpMonitorProfile() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNsxtPolicyLBIcmpMonitorProfileCreate,
+		Read:   resourceNsxtPolicyLBIcmpMonitorProfileRead,
+		Update: resourceNsxtPolicyLBIcmpMonitorProfileUpdate,
+		Delete: resourceNsxtPolicyLBIcmpMonitorProfileDelete,
+		Importer: &schema.ResourceImporter{
+			State: nsxtPolicyPathResourceImporter,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"nsx_id":       getNsxIDSchema(),
+			"path":         getPathSchema(),
+			"display_name": getDisplayNameSchema(),
+			"description":  getDescriptionSchema(),
+			"revision":     getRevisionSchema(),
+			"tag":          getTagsSchema(),
+			"data_length": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Description: "The data size (in byte) of the ICMP healthcheck packet",
+			},
+			"fall_count":   getLbMonitorFallCountSchema(),
+			"interval":     getLbMonitorIntervalSchema(),
+			"monitor_port": getPolicyLbMonitorPortSchema(),
```

Comment:
Does port make sense for icmp monitor? Docs state that for icmp, port is not required.

Reply:
I think it is not required but can be configured via API, on NSX UI we cannot specify port for the Icmp monitor, but we can do this via API. Not sure if we want to keep it or not?

Reply:
I would remove it as it seems inapplicable for icmp